### PR TITLE
Update the messaging for the 'disconnected' dialog

### PR DESCRIPTION
**Pull request by EdmondChuiHW**
# Summary

Quality of life improvement (no semantic changes in this PR).

# Existing

Existing dialog can be dismissed by pressing the icon button.
Doing so will allow users to navigate around the DevTools while disconnected.

screenshot of the existing 'disconnected' dialog

Existing logs can be viewed, even though most features will stop working/appear flaky.
This has been addressed separately by highlighting the disconnected state in the UI https://github.com/facebookexperimental/rn-chrome-devtools-frontend/pull/85:

screenshot of the new 'Reconnect DevTools' button

# Problem

From our internal dogfooding, requests have been raised for the ability to preserve and view logs after disconnection.

This is already supported as discussed above, so it appears to be a discovery problem.

Note: this is a separate issue from some apps' logs being cleared from CDT upon disconnection.

# Test plan

## In this PR

* Remove the icon button for closing the dialog
* Add a new text button for dismissing the dialog
* Add additional texts to explain each of the button's effect.
* Arrange the buttons and texts into vertical stack.

screenshot of the new 'disconnected' dialog with the close button becoming a text button with explainations

# Upstreaming plan

- [ ] This commit should be sent as a patch to the upstream `devtools-frontend` repo. I've reviewed the [contribution guide](https://docs.google.com/document/d/1WNF-KqRSzPLUUfZqQG5AFeU_Ll8TfWYcJasa_XGf7ro/edit#heading=h.9kj7femz1xg5).
- [x] This commit is React Native-specific and cannot be upstreamed.

### CODE REVIEW

File: front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts [hunk 1]
```diff
@@ -27,12 +27,24 @@ const UIStrings = {
...
   /**
    * @description Text on a button to reconnect Devtools when remote debugging terminated.
    * "Remote debugging" here means that DevTools on a PC is inspecting a website running on an actual mobile device
    * (see https://developer.chrome.com/docs/devtools/remote-debugging/).
    */
   reconnectDevtools: 'Reconnect `DevTools`',
+  /**
+   * @description Text in a dialog box to prompt for feedback if the disconnection is unexpected.
+   */
+  sendFeedbackMessage: '[FB-only] Please send feedback if this disconnection is unexpected.',
```
Comment (EdmondChuiHW) [inline front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts:47]: Welcome any alternatives that's more concise!

File: front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts [hunk 2]
```diff
@@ -27,12 +27,24 @@ const UIStrings = {
    * device back in a state where it can be inspected, before DevTools can reconnect to it.
    */
   reconnectWhenReadyByReopening: 'Reconnect when ready by reopening DevTools.',
+  /**
+   * @description Text in a dialog box to explain `DevTools` can still be used while disconnected.
+   */
+  perserveState: 'Dismiss this dialog and continue using `DevTools` while disconnected.',
```
Comment (huntie) [inline front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts:33]: Unnecessary backticks around "DevTools"

File: front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts [hunk 3]
```diff
@@ -27,12 +27,24 @@ const UIStrings = {
    * device back in a state where it can be inspected, before DevTools can reconnect to it.
    */
   reconnectWhenReadyByReopening: 'Reconnect when ready by reopening DevTools.',
```
Comment (huntie) [inline front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts:29]: I think we can also make this slightly more clear.

```suggestion
  reconnectWhenReadyByReopening: 'Reconnect when ready (will reload DevTools).',
```

**Review by huntie — APPROVED** [review]
Love the intention here and agree with this change. Accepting, but I might loop back with another PR improving the visual treatment :)

File: front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts (hunk 2 above)
Comment (EdmondChuiHW) [reply front_end/ui/legacy/RemoteDebuggingTerminatedScreen.ts:33]: CDT has those in some existing strings. I suppose it gets swapped for something else if some build flags are specified? Let's do a pass on all of those and see if they fit our needs.

**Pull request merged**